addons/mail/static/src/models/composer/composer.js

```diff
@@ -574,7 +574,7 @@ function factory(dependencies) {
                     cssClass = "o_channel_redirect";
                 }
                 const mention = this[field].find(mention =>
-                    mention.name === matchName
+                    owl.utils.escape(mention.name) === matchName
```

Comment:
Missing a test for correct channel detection with characters that can be escaped in the name.